### PR TITLE
feat: Nemesis & Dynamic Channels support. Fix: Hud position, CI update

### PR DESCRIPTION
**Pull request by Rushaway**

### CODE REVIEW

File: addons/sourcemod/scripting/TopInfectors.sp [hunk 1]
```diff
@@ -246,21 +284,45 @@ public void Event_OnRoundEnd(Event event, char[] name, bool dontBroadcast)
...
 		}
 	}
 
 	if (g_cvPrint.IntValue <= 0 || g_cvPrint.IntValue == 1)
 		CPrintToChatAll("{darkred}%s", sBuffer);
 
+	bool bDynamicAvailable = false;
+	int iHUDChannel = -1;
+
+#if defined _DynamicChannels_included_
```
Comment (maxime1907) [inline addons/sourcemod/scripting/TopInfectors.sp:316]: put the if just around the function call so before:
```
iHUDChannel = GetDynamicChannel(iChannel);
```

File: addons/sourcemod/scripting/TopInfectors.sp [hunk 2]
```diff
@@ -200,6 +228,16 @@ public void Event_OnClientDeath(Event hEvent, const char[] sEvent, bool bDontBro
 
 		break;
 	}
+	if (g_bNemesis)
+	{
+		int attacker = GetClientOfUserId(hEvent.GetInt("attacker"));
+		if (attacker > 0 && attacker <= MaxClients && IsClientInGame(attacker) && IsPlayerAlive(attacker) && ZR_IsClientZombie(attacker))
+		{
+			if (g_iInfectCount[attacker] == 1) g_iInfectCount[attacker]++; // 1st kill is never counted, so tricky fix to display the correct value..
```
[the remark this answers is not in the document]
Comment (maxime1907) [reply addons/sourcemod/scripting/TopInfectors.sp:236]: why is first kill never counted

File: addons/sourcemod/scripting/TopInfectors.sp [hunk 3]
```diff
@@ -246,21 +284,45 @@ public void Event_OnRoundEnd(Event event, char[] name, bool dontBroadcast)
 		return;
 
 	char sBuffer[512];
-	Format(sBuffer, sizeof(sBuffer), "TOP INFECTORS:");
+	if (g_bNemesis)
+		Format(sBuffer, sizeof(sBuffer), "TOP NEMESIS:");
```
Comment (maxime1907) [inline addons/sourcemod/scripting/TopInfectors.sp:288]: only nemesis/infectors word changes here so put only one format with ternary

File: addons/sourcemod/scripting/TopInfectors.sp [hunk 4]
```diff
@@ -246,21 +284,45 @@ public void Event_OnRoundEnd(Event event, char[] name, bool dontBroadcast)
...
-			Format(sBuffer, sizeof(sBuffer), "%s\n%d. %N - %d INFECTED", sBuffer, i + 1, iSortedList[i][0], iSortedList[i][1]);
-			LogPlayerEvent(iSortedList[i][0], "triggered", i == 0 ? "top_infector" : (i == 1 ? "second_infector" : (i == 2 ? "third_infector" : "super_infector")));
+			if (g_bNemesis)
+			{
+				Format(sBuffer, sizeof(sBuffer), "%s\n%d. %N - %d KILLS", sBuffer, i + 1, iSortedList[i][0], iSortedList[i][1]);
+				LogPlayerEvent(iSortedList[i][0], "triggered", i == 0 ? "top_nemesis" : (i == 1 ? "second_nemesis" : (i == 2 ? "third_nemesis" : "super_nemesis")));
+			}
+			else
+			{
+				Format(sBuffer, sizeof(sBuffer), "%s\n%d. %N - %d INFECTED", sBuffer, i + 1, iSortedList[i][0], iSortedList[i][1]);
```
Comment (maxime1907) [inline addons/sourcemod/scripting/TopInfectors.sp:304]: same lines with only 2 words changing (DRY)

File: addons/sourcemod/scripting/TopInfectors.sp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rushaway) [reply addons/sourcemod/scripting/TopInfectors.sp]: Idk, cant remeber, I also don't understand why the 2 bool for plugins wasnt set in the PR when in local I get them. 😕  
Anyway I tested it why a debug cmd, and now its works correctly.

File: addons/sourcemod/scripting/TopInfectors.sp [hunk 5]
```diff
@@ -246,21 +283,39 @@ public void Event_OnRoundEnd(Event event, char[] name, bool dontBroadcast)
...
+	Format(sBuffer, sizeof(sBuffer), "TOP %s:", g_bNemesis ? "NEMESIS" : "INFECTORS");
 
 	for (int i = 0; i < g_cvAmount.IntValue; i++)
 	{
 		if (iSortedList[i][0])
 		{
 			g_iTopInfector[iSortedList[i][0]] = i;
-			Format(sBuffer, sizeof(sBuffer), "%s\n%d. %N - %d INFECTED", sBuffer, i + 1, iSortedList[i][0], iSortedList[i][1]);
-			LogPlayerEvent(iSortedList[i][0], "triggered", i == 0 ? "top_infector" : (i == 1 ? "second_infector" : (i == 2 ? "third_infector" : "super_infector")));
+			Format(sBuffer, sizeof(sBuffer), "%s\n%d. %N - %d %s", sBuffer, i + 1, iSortedList[i][0], iSortedList[i][1], g_bNemesis ? "KILLS" : "INFECTS");
```
Comment (maxime1907) [inline addons/sourcemod/scripting/TopInfectors.sp:293]: INFECTIONS

**Pull request merged**